apps/frontend/src/app/3_organisms/BorrowLoan/components/ExtendLoanForm/ExtendLoanForm.tsx

```diff
@@ -0,0 +1,387 @@
+  loan: LoanItem;
+  onSuccess: () => void;
+};
+
+export const ExtendLoanForm: FC<ExtendLoanFormProps> = ({ loan }) => {
+  const { checkMaintenance, States } = useMaintenance();
+  const dappLocked = checkMaintenance(States.FULLD2);
+
+  const [nextRolloverDate, setNextRolloverDate] = useState(
+    dayjs(loan.rolloverDate * 1000)
+      .add(1, 'day')
+      .unix(),
+  );
+  const [isCalendarVisible, setIsCalendarVisible] = useState(false);
+
+  const debtToken = useMemo(
+    () => normalizeToken(loan.debtAsset.toLowerCase()),
+    [loan.debtAsset],
+  );
+  const collateralToken = useMemo(
+    () => normalizeToken(loan.collateralAsset.toLowerCase()),
+    [loan.collateralAsset],
+  );
+  const { price: rbtcPrice } = useGetRBTCPrice();
+
+  const [useCollateral, setUseCollateral] = useState(true);
+
+  const { depositAmount, handleSubmit } = useExtendLoan(
+    loan,
+    debtToken,
+    nextRolloverDate,
+    useCollateral,
+  );
+
+  const { borrowPriceUsd, collateralPriceUsd } = useGetCollateralAssetPrice(
+    debtToken,
+    collateralToken,
+  );
+
+  const collateralRatioThresholds = useMemo(
+    () => getCollateralRatioThresholds(collateralToken),
+    [collateralToken],
+  );
+
+  const collateralUsdPrice = useMemo(
+    () =>
+      collateralToken === SupportedTokens.rbtc ? rbtcPrice : collateralPriceUsd,
+    [collateralPriceUsd, collateralToken, rbtcPrice],
+  );
+  const loanTokenUsdPrice = useMemo(
+    () => (debtToken === SupportedTokens.rbtc ? rbtcPrice : borrowPriceUsd),
+    [borrowPriceUsd, debtToken, rbtcPrice],
+  );
+
+  const collateralChange = useMemo(() => {
+    if (useCollateral) {
+      return decimalic(depositAmount)
+        .mul(loanTokenUsdPrice)
+        .div(collateralUsdPrice || 1);
+    }
+    return decimalic(0);
+  }, [collateralUsdPrice, depositAmount, loanTokenUsdPrice, useCollateral]);
+
+  const newCollateralAmount = useMemo(
+    () =>
+      decimalic(loan.collateral.toString()).sub(collateralChange.toString()),
+    [collateralChange, loan.collateral],
+  );
+
+  const newTotalDebt = useMemo(
+    () =>
+      decimalic(loan.debt.toString()).add(useCollateral ? 0 : depositAmount),
+    [depositAmount, loan.debt, useCollateral],
+  );
+
+  const minimumCollateralRatio = useMemo(
+    () =>
+      collateralToken === SupportedTokens.sov
+        ? MINIMUM_COLLATERAL_RATIO_LENDING_POOLS_SOV
+        : MINIMUM_COLLATERAL_RATIO_LENDING_POOLS,
+    [collateralToken],
+  );
+
+  const collateralRatio = useMemo(() => {
+    if (!nextRolloverDate) {
+      return Decimal.from(loan.collateralRatio);
+    }
+    const totalDebtUsd = newTotalDebt.mul(loanTokenUsdPrice);
+
+    return newCollateralAmount
+      .mul(collateralUsdPrice)
+      .div(totalDebtUsd)
+      .mul(100);
+  }, [
+    collateralUsdPrice,
+    loan.collateralRatio,
+    loanTokenUsdPrice,
+    newCollateralAmount,
+    newTotalDebt,
+    nextRolloverDate,
+  ]);
+
+  const collateralAssetPrice = useMemo(() => {
```

Comment:
This is not correct because you're assuming that either `collateralToken` or `debtToken` is `RBTC`. What if you want to borrow `DLLR` with `SOV` as collateral? Look at `AdjustLoanForm` how the calculation should look like (and you can create a shared util function for it).